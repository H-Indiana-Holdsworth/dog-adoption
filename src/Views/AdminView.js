import React from 'react';
import { useState } from 'react/cjs/react.development';
import DogForm from '../Components/DogForm';
import { createDog } from '../services/service-dogs';

export default function AdminView() {
  const [dog, setDog] = useState({});

  const updateDogState = (key, value) => {
    dog[key] = value;
    setDog({ ...dog });
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    await createDog(dog);
  };

  return (
    <div>
      <DogForm {...dog} handleSubmit={handleSubmit} updateDogState={updateDogState} />
    </div>
  );
}
