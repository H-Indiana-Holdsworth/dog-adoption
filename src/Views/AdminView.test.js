import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import AdminView from './AdminView';

test('should render container', () => {
  const { container } = render(
    <MemoryRouter>
      <AdminView />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
