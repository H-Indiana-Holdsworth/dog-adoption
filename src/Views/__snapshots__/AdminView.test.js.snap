// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render container 1`] = `
<div>
  <div>
    <header>
      <a
        aria-current="page"
        class="home-link active"
        href="/"
      >
        Home
      </a>
    </header>
    <div
      class="body"
    >
      <form>
        <label>
          Name:
          <input
            name="name"
            type="text"
            value=""
          />
        </label>
        <label>
          Image URL:
          <input
            name="image"
            type="text"
            value=""
          />
        </label>
        <label>
          Bio:
          <input
            class="bio"
            name="bio"
            type="text"
            value=""
          />
        </label>
        <label>
          Breed:
          <input
            name="breed"
            type="text"
            value=""
          />
        </label>
        <label>
          Age:
          <input
            name="age"
            type="text"
            value=""
          />
        </label>
        <button>
          Save
        </button>
      </form>
    </div>
  </div>
</div>
`;
